File: test/demo/wslc-polyglot-hello/polyglot/servers/srv.R
port <- as.integer(commandArgs(TRUE)[1])

srv <- serverSocket(port)
con <- socketAccept(srv, blocking = TRUE, open = "r+b")
readLines(con, n = 1, warn = FALSE)            # Windows request
writeLines("hello world from [r]", con)
readLines(con, n = 1, warn = FALSE)            # Windows ack -> exit
close(con)
close(srv)

File: test/demo/wslc-polyglot-hello/polyglot/servers/srv.php
<?php
$port = (int)$argv[1];

$server = stream_socket_server("tcp://0.0.0.0:$port");
$conn = stream_socket_accept($server, -1);
fread($conn, 64);                              // Windows request
fwrite($conn, "hello world from [php]\n");
fread($conn, 64);                              // Windows ack -> exit
fclose($conn);
